resource-managers/yarn/src/test/java/org/apache/hadoop/net/ServerSocketUtil.java

```diff
@@ -0,0 +1,132 @@
+/*
```

Comment:
Add this class. otherwise:
```
[info] YarnClusterSuite:
[info] org.apache.spark.deploy.yarn.YarnClusterSuite *** ABORTED *** (33 milliseconds)
[info]   java.lang.NoClassDefFoundError: org/apache/hadoop/net/ServerSocketUtil
[info]   at org.apache.hadoop.yarn.server.MiniYARNCluster.serviceInit(MiniYARNCluster.java:260)
```
I try to do it by maven, but failed. It seems that [`sbt-pom-reader` does not support `test-jar`](https://github.com/sbt/sbt-pom-reader/pull/#14) very well.:

```xml
    <dependency>
      <groupId>org.apache.hadoop</groupId>
      <artifactId>hadoop-common</artifactId>
      <version>${hadoop.version}</version>
      <type>test-jar</type>
      <scope>test</scope>
      <exclusions>
        <exclusion>
          <groupId>*</groupId>
          <artifactId>*</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
```

Reply:
We do have other test-jar imports; some use `type` and some `classifier`. Have you tried the latter? e.g. we have

```
    <dependency>
      <groupId>org.apache.hadoop</groupId>
      <artifactId>hadoop-yarn-server-tests</artifactId>
      <classifier>tests</classifier>
      <scope>test</scope>
    </dependency>
```

Reply:
Yes, I tried it, this can be successful on hadoop-3.2, but throws a compilation exception on hadoop-2.7(It seems exclude `hadoop-common`).
```java
[error] /Users/yumwang/SPARK-20845/spark/resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ApplicationMasterSuite.scala:34: Class org.apache.hadoop.conf.Configuration not found - continuing with a stub.
[error]     val yarnConf = new YarnConfiguration()
[error] 
```